packages/smooth_app/lib/pages/product/confirm_and_upload_picture.dart

```diff
@@ -0,0 +1,116 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:smooth_app/generic_lib/buttons/smooth_action_button.dart';
+import 'package:smooth_app/helpers/picture_capture_helper.dart';
+import 'package:smooth_app/helpers/ui_helpers.dart';
+
+class ConfirmAndUploadPicture extends StatefulWidget {
+  const ConfirmAndUploadPicture({
+    required this.barcode,
+    required this.imageType,
+    required this.initialPhoto,
+  });
+
+  final ImageField imageType;
+  final String barcode;
+  final File initialPhoto;
+
+  @override
+  State<ConfirmAndUploadPicture> createState() =>
+      _ConfirmAndUploadPictureState();
+}
+
+class _ConfirmAndUploadPictureState extends State<ConfirmAndUploadPicture> {
+  late File photo;
+
+  @override
+  void initState() {
+    super.initState();
+    photo = widget.initialPhoto;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    // Picture is captured, show it to the user one last time and ask for
+    // confirmation before uploading. Also present an option to retake the
+    // picture as sometimes the picture can be blurry.
+    return Scaffold(
+      backgroundColor: Colors.black,
+      appBar: AppBar(),
+      body: Stack(
+        children: <Widget>[
+          Positioned(
+            child: Align(
+              alignment: Alignment.center,
+              child: Image.file(photo),
+            ),
+          ),
+          Positioned(
+            child: Align(
+              alignment: Alignment.bottomCenter,
+              child: Padding(
+                padding: const EdgeInsets.only(bottom: MEDIUM_SPACE),
+                child: Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceAround,
+                  children: <Widget>[
+                    SmoothActionButton(
+                        text: appLocalizations.retake_photo_button_label,
+                        onPressed: () async {
+                          final File? retakenPhoto = await pickImageAndCrop();
+                          if (retakenPhoto == null) {
+                            // User chose not to upload the image.
+                            Navigator.pop(context);
+                            return;
+                          }
```

Comment:
I don't get it: why do you exit the page? The user wanted to pick image and crop, and that was not satisfactory, that's it. The photo is still valid.

Reply:
So it's like the user uploaded a picture, then decided he doesn't want to upload anymore (clicked on retake and then exited the camera). I'm treating a click on retake as "the last picture was bad, discard it and take a new one", and if the user decides to exit the camera from the then we just take them back to add new product page, from where they have another chance to upload a new image if they want to. 🤷